### PR TITLE
Add compression level for data export

### LINKED ISSUE #38412: Allow to specify compression level for data export.
**Use case**

We allow to export data in gz, xz, bz2, lz4, and zst, but only with predefined compression level.
Sometimes I want higher compression.

### PR DESCRIPTION
**Pull request by evillique**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add the ability to specify compression level during data export.

**Syntax:**
```SELECT * INTO OUTFILE 'file.lz4' COMPRESSION 'lz4' LEVEL <n>```, where `n` is a positive integer in certain valid range for specific compression algorithm.

Closes #38412

### CODE REVIEW

File: src/Client/ClientBase.cpp [hunk 1]
```diff
@@ -525,16 +525,26 @@ try
...
                 {
                     const auto & compression_method_node = query_with_output->compression->as<ASTLiteral &>();
                     compression_method = compression_method_node.value.safeGet<std::string>();
+
+                    if (query_with_output->compression_level)
+                    {
+                        const auto & compression_level_node = query_with_output->compression_level->as<ASTLiteral &>();
+                        bool res = compression_level_node.value.tryGet<UInt64>(compression_level);
+
+                        if (!res || compression_level < 1 || compression_level > 9)
```
Comment (CurtizJ) [inline src/Client/ClientBase.cpp:539]: Why only from 1 to 9? E.g. lz4 support compression levels from 1 to 12 and zstd from 1 to 22.

**Comment by alexey-milovidov** [comment]
The syntax `COMPRESSION 'gzip' 1` looks unnatural.
Maybe make it `COMPRESSION 'gzip' LEVEL 1`?

File: src/IO/CompressionMethod.h [hunk 2]
```diff
@@ -46,6 +46,9 @@ std::string toContentEncodingName(CompressionMethod method);
   */
 CompressionMethod chooseCompressionMethod(const std::string & path, const std::string & hint);
 
+/// Get a range of the valid compression levels for the compression method.
+std::pair<uint64_t, uint64_t> getCompressionLevelRange(const CompressionMethod & method);
```
Comment (alexey-milovidov) [inline src/IO/CompressionMethod.h:50]: Some algorithms may have negative levels.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
LGTM, minor improvements are possible, but let's merge now :)

**Comment by alexey-milovidov** [comment]
How compression level will be specified on data export with table functions?

**Comment by evillique** [comment]
> How compression level will be specified on data export with table functions?

It does sound a little bit harder since as I can see compression format is automatically inferred from the path and there is no separate way to specify it. Without adding additional arguments to these functions compression level will also be needed to be inferred from it.

It probably could be added in the path field after some special character like `:` e.g. `file('path.lz4:9')`, but I'm not sure about it.

**Pull request merged**